src/OrchardCore.Modules/OrchardCore.Markdown/Drivers/MarkdownBodyPartDisplay.cs

```diff
@@ -73,9 +76,8 @@ private async ValueTask BuildViewModel(MarkdownBodyPartViewModel model, Markdown
             model.ContentItem = MarkdownBodyPart.ContentItem;
             templateContext.SetValue("Model", model);
 
-            var markdown = await _liquidTemplatemanager.RenderAsync(MarkdownBodyPart.Markdown, System.Text.Encodings.Web.HtmlEncoder.Default, templateContext);
-
-            model.Html = Markdig.Markdown.ToHtml(markdown ?? "");
+            model.Markdown = await _liquidTemplatemanager.RenderAsync(MarkdownBodyPart.Markdown, NullEncoder.Default, templateContext);
+            model.Html = Markdig.Markdown.ToHtml(model.Markdown ?? "");
```

Comment:
Looks right to use the `NullEncoder` to initialize the `Markdown` field

But then, to set the `Html` field maybe we need

    model.Html = _htmlEncoder.Encode(Markdig.Markdown.ToHtml(model.Markdown ?? "");

Here, assuming that `Markdig.Markdown.ToHtml()` doesn't do any html encoding.
And knowing that the related shape do `@Html.Raw(Model.Html)` <= but maybe wrong here

Note: For the html body part we use `_htmlEncoder` and the shape also uses `@Html.Raw(Model.Html)`.

Reply:
Liquid should use the html encoder here. And we should not encode the markdown.